### ISSUE TITLE
docker API invocation  returns "code":"does_not_exist","message":"Does Not Exist" 

### ISSUE THREAD

**Issue by kritiagg** [zaproxy/zaproxy#2217]
Hi, 
I ran ZAP docker using the following command: 
docker run -p 8090:8090 -i owasp/zap2docker-stable zap.sh -daemon -port 8090 -config api.key=12345 -host 0.0.0.0

I tried running the following request:
http://0.0.0.0:8090/JSON/spider/action/scan/?zapapiformat=JSON&apikey=12345&url=http%3A%2F%2Fwww.google.com&maxChildren=4&recurse=&contextName=default

But the output says:
"code":"does_not_exist","message":"Does Not Exist" 

The same output is returned for all the other api calls like ascan etc.

**Comment by thc202** [comment]
That error happens when trying to spider a context that does not exist. Do you have a context named "default"?

If you are making the requests manually (as opposed to use one the ZAP API clients), you need to remove the parameter "contextName" if you don't want to spider a context (see #2108 and #2122).

**Comment by kritiagg** [comment]
I understand, thanks. 
Also, if I try to request:
http://0.0.0.0:8090/JSON/ascan/action/scan/?zapapiformat=JSON&apikey=12345&url=http%3A%2F%2Fwww.google.com&recurse=true&inScopeOnly=true&scanPolicyName=&method=&postData=
I get the following error:
{"code":"url_not_found","message":"URL Not Found in the Scan Tree"}

**Comment by kritiagg** [comment]
Thanks, I understand.

Though I tried doing ascan without spidering, and it is giving the 'url not part of the scan tree' error.
I perceive that its essential to do spidering before doing active scan. But how can I pass the scan results using the ascan api?

**Comment by thc202** [comment]
Right, you need to proxy/send the requests (manually, spider or by other means) before starting the active scanner.

What do you mean by "pass"? Check which alerts were raised? For that you can use the "core" view "alerts".

**Comment by kingthorin** [comment]
@kritiagg resolved?

**Comment by kritiagg** [comment]
Yes, Thanks :)

On Mon, Feb 15, 2016 at 7:07 AM, kingthorin <notifications@github.com>
wrote:

> @kritiagg <https://github.com/kritiagg> resolved?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zaproxy/zaproxy/issues/2217#issuecomment-184021638>.
>



-- 
Thanks and regards
Kriti
Security Researcher
Adobe Secure Software Engineering Team
Adobe Systems, India
